[Automation] native-image run fails for software.amazon.awssdk:sdk-core:2.42.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.42.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:sdk-core:2.37.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26082529291/job/76688415704
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
ve-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=software.amazon.awssdk.sdk-core_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/sdk-core/2.37.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/annotations/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-client-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/utils/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.reactivestreams/reactive-streams/1.0.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.slf4j/slf4j-api/1.7.36,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/metrics-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/endpoints-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-auth-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/checksums-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/identity-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-auth-aws/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/checksums/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/profiles/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/retries-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/retries/2.34.0 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=800abc46-c76c-f1c6-62dd-5feecb738d65 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'software.amazon.awssdk.sdk-core_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (5.9s @ 0.13GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (40.1s @ 1.16GB)
   10,274 types,  12,294 fields, and  46,432 methods found reachable
    3,387 types,     277 fields, and   2,981 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (5.8s @ 1.05GB)
[4/8] Parsing methods...      [**]                                                                       (3.8s @ 1.14GB)
[5/8] Inlining methods...     [***]                                                                      (2.4s @ 1.22GB)
[6/8] Compiling methods...    [*****]                                                                   (22.4s @ 0.85GB)
[7/8] Laying out methods...   [**]                                                                       (3.6s @ 1.11GB)
[8/8] Creating image...       [**]                                                                       (4.1s @ 1.24GB)
  14.01MB (38.91%) for code area:    29,677 compilation units
  17.24MB (47.89%) for image heap:  270,025 objects and 162 resources
   4.75MB (13.20%) for other data
  35.99MB in total image size, 33.36MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   7.98MB java.base                                            4.19MB byte[] for code metadata
   2.78MB svm.jar (Native Image)                               3.18MB byte[] for java.lang.String
 560.23kB assertj-core-3.22.0.jar                              1.70MB java.lang.String
 483.00kB java.xml                                             1.32MB java.lang.Class
 236.25kB java.rmi                                             1.08MB byte[] for general heap data
 198.29kB junit-jupiter-engine-5.9.2.jar                       1.07MB com.oracle.svm.core.hub.DynamicHubCompanion
 124.42kB java.naming                                        470.91kB java.util.concurrent.ConcurrentHashMap$Node
 122.96kB junit-platform-launcher-1.9.2.jar                  323.36kB java.util.HashMap$Node
 106.86kB java.logging                                       303.22kB java.lang.Object[]
  99.62kB junit-platform-engine-1.9.2.jar                    300.67kB c.o.svm.core.hub.DynamicHub$ReflectionMetadata
 859.66kB for 36 more packages                                 3.30MB for 2689 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 25 component(s) is embedded in binary (1.16kB). 10 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        7.4s (8.1% of total time) in 982 GCs | Peak RSS: 2.06GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/software.amazon.awssdk.sdk-core_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'software.amazon.awssdk.sdk-core_tests-tests' in 1m 30s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassUsesThreadContextClassLoaderBeforeProvidedClassLoaders() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassUsesProvidedClassLoaderBeforeContextLoader() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassFallsBackToClassForNameAfterTryingOtherLoaders() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClasspathInterceptorChainFactoryTest > getInterceptorsLoadsExecutionInterceptorsDeclaredInClasspathResource() SUCCESSFUL

software_amazon_awssdk.sdk_core.Crc32CChecksumTest > markAndResetRestoreTheDelegatedChecksumState() FAILED

software_amazon_awssdk.sdk_core.Crc32ChecksumTest > markAndResetRestoreTheDelegatedChecksumState() SUCCESSFUL

software_amazon_awssdk.sdk_core.MimetypeTest > getInstanceInitializesClasspathBackedMimetypeRegistry() SUCCESSFUL

software_amazon_awssdk.sdk_core.SystemPropertyHttpServiceProviderTest > defaultSyncClientBuilderInstantiatesClassNamedBySystemProperty() SUCCESSFUL


Failures (1):
  JUnit Jupiter:Crc32CChecksumTest:markAndResetRestoreTheDelegatedChecksumState()
    MethodSource [className = 'software_amazon_awssdk.sdk_core.Crc32CChecksumTest', methodName = 'markAndResetRestoreTheDelegatedChecksumState', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke constructor 'public java.util.zip.CRC32C()'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "java.util.zip.CRC32C",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:97)
       java.base@25.0.3/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:496)
       java.base@25.0.3/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.lambda$createJavaCrc32C$1(ChecksumProvider.java:171)
       java.base@25.0.3/java.util.Optional.map(Optional.java:260)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.createJavaCrc32C(ChecksumProvider.java:169)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.crc32cImplementation(ChecksumProvider.java:90)
       software.amazon.awssdk.checksums.SdkChecksum.forAlgorithm(SdkChecksum.java:41)
       software.amazon.awssdk.core.internal.checksums.LegacyDelegatingChecksum.<init>(LegacyDelegatingChecksum.java:30)
       [...]

Test run finished after 6 ms
[         8 containers found      ]
[         0 containers skipped    ]
[         8 containers started    ]
[         0 containers aborted    ]
[         8 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/reports/problems/problems-report.html
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/software.amazon.awssdk.sdk-core_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD FAILED in 1m 34s
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for software.amazon.awssdk:sdk-core:2.37.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 35s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.42.0 at 2026-05-19T08:02:39Z with exit code 1 after 95s
❌ ERROR: 'native-image run' for 2.42.0 failed with exit code 1.
FAILED[native-image run][2.42.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.42.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:sdk-core:2.37.0"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.42.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:sdk-core:2.37.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26082529291/job/76688415740
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
data/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/sdk-core/2.37.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/annotations/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-client-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/utils/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.reactivestreams/reactive-streams/1.0.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.slf4j/slf4j-api/1.7.36,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/metrics-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/endpoints-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-auth-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/checksums-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/identity-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-auth-aws/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/checksums/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/profiles/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/retries-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/retries/2.34.0 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=25182020-6653-bca6-1c1a-691b111769d6 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'software.amazon.awssdk.sdk-core_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.0s @ 0.20GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (45.0s @ 1.18GiB)
   10,433 types,  12,408 fields, and  47,029 methods found reachable
      530 types,     102 fields, and   1,573 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
      162 resource accesses registered with 262.58KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.1s @ 1.04GiB)
[4/8] Parsing methods...      [**]                                                                      (4.5s @ 1.11GiB)
[5/8] Inlining methods...     [***]                                                                     (2.4s @ 1.18GiB)
[6/8] Compiling methods...    [*****]                                                                  (26.3s @ 0.85GiB)
[7/8] Laying out methods...   [**]                                                                      (3.9s @ 1.08GiB)
[8/8] Creating image...       [**]                                                                      (4.3s @ 1.27GiB)
  12.44MiB (37.77%) for code area:    30,085 compilation units
  17.38MiB (52.77%) for image heap:  281,567 objects and 24 resources
   3.11MiB ( 9.45%) for other data
  32.92MiB in total image size, 30.38MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.36MiB o.g.n.runtime/com.oracle.svm.core.code              3.89MiB byte[] for code metadata
 872.01KiB java.base/sun.security.ssl                          3.14MiB byte[] for string data
 768.52KiB java.base/java.util                                 1.70MiB java.base/java.lang.String
 464.15KiB java.base/com.sun.crypto.provider                   1.27MiB java.base/java.lang.Class
 414.69KiB java.base/jdk.internal.classfile.impl               1.11MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 345.69KiB java.base/java.lang                               973.15KiB byte[] for general heap data
 344.07KiB java.base/java.lang.invoke                        484.38KiB byte[] for reflection metadata
 279.89KiB java.base/java.util.concurrent                    433.28KiB heap alignment
 276.81KiB java.base/sun.security.provider                   405.12KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 252.95KiB java.base/java.io                                 329.28KiB java.base/java.util.HashMap$Node
   7.15MiB for 370 more packages                               3.69MiB for 2717 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 24 component(s) is embedded in binary (1.18KiB). 10 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       7.5s (7.2% of total time) in 1029 GCs | Peak RSS: 1.90GiB | CPU load: 3.65
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/software.amazon.awssdk.sdk-core_tests-tests (executable, 30.38MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'software.amazon.awssdk.sdk-core_tests-tests' in 1m 42s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassUsesThreadContextClassLoaderBeforeProvidedClassLoaders() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassUsesProvidedClassLoaderBeforeContextLoader() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassFallsBackToClassForNameAfterTryingOtherLoaders() SUCCESSFUL
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/software.amazon.awssdk.sdk-core_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.



software_amazon_awssdk.sdk_core.ClasspathInterceptorChainFactoryTest > getInterceptorsLoadsExecutionInterceptorsDeclaredInClasspathResource() SUCCESSFUL

software_amazon_awssdk.sdk_core.Crc32CChecksumTest > markAndResetRestoreTheDelegatedChecksumState() FAILED

software_amazon_awssdk.sdk_core.Crc32ChecksumTest > markAndResetRestoreTheDelegatedChecksumState() SUCCESSFUL

software_amazon_awssdk.sdk_core.MimetypeTest > getInstanceInitializesClasspathBackedMimetypeRegistry() SUCCESSFUL

software_amazon_awssdk.sdk_core.SystemPropertyHttpServiceProviderTest > defaultSyncClientBuilderInstantiatesClassNamedBySystemProperty() SUCCESSFUL


Failures (1):
  JUnit Jupiter:Crc32CChecksumTest:markAndResetRestoreTheDelegatedChecksumState()
    MethodSource [className = 'software_amazon_awssdk.sdk_core.Crc32CChecksumTest', methodName = 'markAndResetRestoreTheDelegatedChecksumState', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke constructor 'public java.util.zip.CRC32C()'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "java.util.zip.CRC32C",
      "methods": [
        {
          "name": "<init>",
          "parameterTypes": []
        }
      ]
    }

BUILD FAILED in 1m 45s
The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.2/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:101)
       java.base@25.0.2/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:496)
       java.base@25.0.2/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.lambda$createJavaCrc32C$1(ChecksumProvider.java:171)
       java.base@25.0.2/java.util.Optional.map(Optional.java:260)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.createJavaCrc32C(ChecksumProvider.java:169)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.crc32cImplementation(ChecksumProvider.java:90)
       software.amazon.awssdk.checksums.SdkChecksum.forAlgorithm(SdkChecksum.java:41)
       software.amazon.awssdk.core.internal.checksums.LegacyDelegatingChecksum.<init>(LegacyDelegatingChecksum.java:30)
       [...]

Test run finished after 7 ms
[         8 containers found      ]
[         0 containers skipped    ]
[         8 containers started    ]
[         0 containers aborted    ]
[         8 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for software.amazon.awssdk:sdk-core:2.37.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 47s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.42.0 at 2026-05-19T08:05:32Z with exit code 1 after 108s
❌ ERROR: 'native-image run' for 2.42.0 failed with exit code 1.
FAILED[native-image run][2.42.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.42.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:sdk-core:2.37.0"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.42.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:sdk-core:2.37.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26082529291/job/76688415844
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
achability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/sdk-core/2.37.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/annotations/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-client-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/utils/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.reactivestreams/reactive-streams/1.0.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.slf4j/slf4j-api/1.7.36,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/metrics-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/endpoints-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-auth-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/checksums-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/identity-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/http-auth-aws/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/checksums/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/profiles/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/retries-spi/2.34.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/software.amazon.awssdk/retries/2.34.0 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=f9f62c90-da8b-27fa-7dc8-289f05b42fa9 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'software.amazon.awssdk.sdk-core_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.2s @ 0.20GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (46.8s @ 1.18GiB)
   10,532 types,  12,698 fields, and  47,876 methods found reachable
      548 types,     102 fields, and   1,585 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
      164 resource accesses registered with 262.73KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.6s @ 1.05GiB)
[4/8] Parsing methods...      [**]                                                                      (4.9s @ 1.13GiB)
[5/8] Inlining methods...     [***]                                                                     (2.6s @ 1.18GiB)
[6/8] Compiling methods...    [*****]                                                                  (28.2s @ 0.98GiB)
[7/8] Laying out methods...   [**]                                                                      (4.3s @ 1.21GiB)
[8/8] Creating image...       [**]                                                                      (4.2s @ 1.37GiB)
  12.75MiB (37.66%) for code area:    30,728 compilation units
  17.94MiB (52.99%) for image heap:  292,475 objects and 26 resources
   3.16MiB ( 9.34%) for other data
  33.85MiB in total image size, 31.25MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.39MiB o.g.n.runtime/com.oracle.svm.core.code              4.00MiB byte[] for code metadata
 887.83KiB java.base/sun.security.ssl                          3.28MiB byte[] for string data
 770.70KiB java.base/java.util                                 1.74MiB java.base/java.lang.String
 464.32KiB java.base/com.sun.crypto.provider                   1.28MiB java.base/java.lang.Class
 414.69KiB java.base/jdk.internal.classfile.impl               1.12MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 354.93KiB java.base/java.lang                               961.53KiB byte[] for general heap data
 344.07KiB java.base/java.lang.invoke                        502.38KiB byte[] for reflection metadata
 280.01KiB java.base/java.util.concurrent                    420.62KiB heap alignment
 276.81KiB java.base/sun.security.provider                   408.98KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 260.86KiB java.base/java.io                                 361.81KiB java.base/java.lang.Object[]
   7.40MiB for 373 more packages                               3.91MiB for 2724 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 24 component(s) is embedded in binary (1.18KiB). 10 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       7.8s (7.2% of total time) in 1073 GCs | Peak RSS: 2.02GiB | CPU load: 3.66
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/software.amazon.awssdk.sdk-core_tests-tests (executable, 31.25MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'software.amazon.awssdk.sdk-core_tests-tests' in 1m 47s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".----------------------------------------


SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassUsesThreadContextClassLoaderBeforeProvidedClassLoaders() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassUsesProvidedClassLoaderBeforeContextLoader() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClassLoaderHelperTest > loadClassFallsBackToClassForNameAfterTryingOtherLoaders() SUCCESSFUL

software_amazon_awssdk.sdk_core.ClasspathInterceptorChainFactoryTest > getInterceptorsLoadsExecutionInterceptorsDeclaredInClasspathResource() SUCCESSFUL

software_amazon_awssdk.sdk_core.Crc32CChecksumTest > markAndResetRestoreTheDelegatedChecksumState() FAILED

software_amazon_awssdk.sdk_core.Crc32ChecksumTest > markAndResetRestoreTheDelegatedChecksumState() SUCCESSFUL

software_amazon_awssdk.sdk_core.MimetypeTest > getInstanceInitializesClasspathBackedMimetypeRegistry() SUCCESSFUL

software_amazon_awssdk.sdk_core.SystemPropertyHttpServiceProviderTest > defaultSyncClientBuilderInstantiatesClassNamedBySystemProperty() SUCCESSFUL


Failures (1):
  JUnit Jupiter:Crc32CChecksumTest:markAndResetRestoreTheDelegatedChecksumState()
    MethodSource [className = 'software_amazon_awssdk.sdk_core.Crc32CChecksumTest', methodName = 'markAndResetRestoreTheDelegatedChecksumState', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke constructor 'public java.util.zip.CRC32C()'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "java.util.zip.CRC32C",
      "methods": [
        {
          "name": "<init>",
          "parameterTypes": []
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.2/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:101)
       java.base@25.0.2/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:496)
       java.base@25.0.2/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.lambda$createJavaCrc32C$1(ChecksumProvider.java:171)
       java.base@25.0.2/java.util.Optional.map(Optional.java:260)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.createJavaCrc32C(ChecksumProvider.java:169)
       software.amazon.awssdk.checksums.internal.ChecksumProvider.crc32cImplementation(ChecksumProvider.java:90)
       software.amazon.awssdk.checksums.SdkChecksum.forAlgorithm(SdkChecksum.java:41)
       software.amazon.awssdk.core.internal.checksums.LegacyDelegatingChecksum.<init>(LegacyDelegatingChecksum.java:30)
       [...]

Test run finished after 4 ms
[         8 containers found      ]
[         0 containers skipped    ]
[         8 containers started    ]
[         0 containers aborted    ]
[         8 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         1 tests failed          ]



[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/software.amazon.awssdk/sdk-core/2.37.0/build/native/nativeTestCompile/software.amazon.awssdk.sdk-core_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 51s
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for software.amazon.awssdk:sdk-core:2.37.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 53s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.42.0 at 2026-05-19T08:05:42Z with exit code 1 after 113s
❌ ERROR: 'native-image run' for 2.42.0 failed with exit code 1.
FAILED[native-image run][2.42.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.42.0" ./gradlew clean test -Pcoordinates="software.amazon.awssdk:sdk-core:2.37.0"]
```

--- a/metadata/software.amazon.awssdk/sdk-core/2.42.0/reachability-metadata.json
+++ b/metadata/software.amazon.awssdk/sdk-core/2.42.0/reachability-metadata.json
@@ -1,0 +1,48 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "software.amazon.awssdk.core.checksums.Crc32CChecksum"
+      },
+      "type" : "java.util.zip.CRC32C",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "software.amazon.awssdk.core.interceptor.ClasspathInterceptorChainFactory"
+      },
+      "type" : "software.amazon.awssdk.core.internal.interceptor.HttpChecksumValidationInterceptor",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "software.amazon.awssdk.core.internal.util.ClassLoaderHelper"
+      },
+      "type" : "software.amazon.awssdk.core.internal.interceptor.HttpChecksumValidationInterceptor"
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "software.amazon.awssdk.core.checksums.Crc32CChecksum"
+      },
+      "glob" : "org/slf4j/impl/StaticLoggerBinder.class"
+    },
+    {
+      "condition" : {
+        "typeReached" : "software.amazon.awssdk.core.internal.util.Mimetype"
+      },
+      "glob" : "software/amazon/awssdk/core/util/mime.types"
+    }
+  ]
+}

--- a/metadata/software.amazon.awssdk/sdk-core/index.json
+++ b/metadata/software.amazon.awssdk/sdk-core/index.json
@@ -1,6 +1,22 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.42.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/software/amazon/awssdk/sdk-core/$version$/sdk-core-$version$-sources.jar",
+    "repository-url" : "https://github.com/aws/aws-sdk-java-v2",
+    "test-code-url" : "https://github.com/aws/aws-sdk-java-v2/tree/$version$/core/sdk-core/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/software/amazon/awssdk/sdk-core/$version$/sdk-core-$version$-javadoc.jar",
+    "description" : "AWS SDK for Java v2 SDK Core provides runtime classes and shared infrastructure used by AWS service clients to make requests, configure clients, transform payloads, handle responses, retries, pagination, checksums, and exceptions. It is a foundational JVM module for AWS SDK service artifacts rather than a standalone service client.",
+    "test-version" : "2.37.0",
+    "tested-versions" : [
+      "2.42.0"
+    ],
+    "allowed-packages" : [
+      "software.amazon.awssdk.core",
+      "software.amazon.awssdk.checksums.internal"
+    ]
+  },
+  {
     "metadata-version" : "2.37.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/software/amazon/awssdk/sdk-core/$version$/sdk-core-$version$-sources.jar",
     "repository-url" : "https://github.com/aws/aws-sdk-java-v2",

--- a/stats/software.amazon.awssdk/sdk-core/2.42.0/stats.json
+++ b/stats/software.amazon.awssdk/sdk-core/2.42.0/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "2.42.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 6,
+          "totalCalls" : 6
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 8,
+      "totalCalls" : 8
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 1102,
+        "missed" : 44510,
+        "ratio" : 0.02416,
+        "total" : 45612
+      },
+      "line" : {
+        "covered" : 209,
+        "missed" : 10857,
+        "ratio" : 0.018887,
+        "total" : 11066
+      },
+      "method" : {
+        "covered" : 55,
+        "missed" : 3672,
+        "ratio" : 0.014757,
+        "total" : 3727
+      }
+    }
+  } ]
+}

--- a/tests/src/software.amazon.awssdk/sdk-core/2.37.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/software.amazon.awssdk/sdk-core/2.37.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -5,6 +5,36 @@
         "typeReached" : "software.amazon.awssdk.core.internal.http.loader.SystemPropertyHttpServiceProvider"
       },
       "type" : "software_amazon_awssdk.sdk_core.SystemPropertyHttpServiceProviderTest$TestSdkHttpService"
+    },
+    {
+      "condition" : {
+        "typeReached" : "software_amazon_awssdk.sdk_core.ClassLoaderHelperTest"
+      },
+      "type" : "example.missing.ClassLoaderHelperTarget"
+    },
+    {
+      "condition" : {
+        "typeReached" : "software_amazon_awssdk.sdk_core.ClassLoaderHelperTest"
+      },
+      "type" : "java.lang.Integer"
+    },
+    {
+      "condition" : {
+        "typeReached" : "software_amazon_awssdk.sdk_core.ClassLoaderHelperTest"
+      },
+      "type" : "java.lang.String"
+    },
+    {
+      "condition" : {
+        "typeReached" : "software.amazon.awssdk.core.internal.http.loader.SystemPropertyHttpServiceProvider"
+      },
+      "type" : "software_amazon_awssdk.sdk_core.SystemPropertyHttpServiceProviderTest$TestSdkHttpService",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
     }
   ],
   "resources" : [
@@ -13,6 +43,18 @@
         "typeReached" : "software.amazon.awssdk.core.interceptor.ClasspathInterceptorChainFactory"
       },
       "glob" : "sdk-core-test/execution.interceptors"
+    },
+    {
+      "condition" : {
+        "typeReached" : "software_amazon_awssdk.sdk_core.ClassLoaderHelperTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "software_amazon_awssdk.sdk_core.ClassLoaderHelperTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
     }
   ]
 }

